File: calculateMode.js
function calculateMode(array){
let occurances = {};
let answer = [];
let maxVal = 0;
for(let i = 0; i < array.length; i++){
let element = array[i];
  if(occurances[element]){
    occurances[element] += 1;
    maxVal = occurances[element];
  } else {
    occurances[element] = 1;
    maxVal = occurances[element]
  } 
}
  for (key in occurances) {
    if(occurances[key] === maxVal){
      answer.push(key);
    } 
  }
  return answer
}
console.log(calculateMode(["who", "what", "where",'who']))
console.log(calculateMode([1,1,2,2]))

File: calculate_mode.py
def calculate_mode(list):
  occurances = {}
  max_val = 0
  answer = []
  for i in list:
    if i not in occurances:
        occurances[i] = 1
        max_val = occurances[i]
    else:
        occurances[i] += 1
        max_val = occurances[i]
  for key in occurances:
    if(occurances[key] == max_val):
        answer.append(key)
    
  print(answer)
  
calculate_mode([1,2,3,3,3])
